src/pudl/extract/gridpathratoolkit.py

```diff
@@ -0,0 +1,128 @@
+    produce the time series based on the wind and solar capacity aggregation tables.
+    """
+
+    def _extract_csv(part: str, ds: Datastore) -> pd.DataFrame:
+        return pd.read_csv(
+            BytesIO(ds.get_unique_resource("gridpathratoolkit", part=part))
+        )
+
+    def _extract_capacity_factor(part: str, ds: Datastore) -> pd.DataFrame:
+        zf = ds.get_zipfile_resource("gridpathratoolkit", part=part)
+        profiles = pd.read_csv(BytesIO(zf.read("timestamps.csv"))).rename(
+            columns={"HE": "hour"}
+        )
+
+        files = [fn for fn in zf.namelist() if "timestamps" not in fn]
+        for fn in files:
+            aggregation_key = Path(fn).stem
+            new_profile = pd.read_csv(
+                BytesIO(zf.read(fn)), header=None, names=[aggregation_key]
+            )
+            profiles = pd.concat([profiles, new_profile], axis="columns")
+        return profiles
+
+    @asset(
+        name=f"raw_gridpathratoolkit__{part}",
+        required_resource_keys={"datastore", "dataset_settings"},
+    )
+    def _extract(context):
+        """Extract raw GridPath RA Toolkit renewable energy generation profiles.
+
+        Args:
+            context: dagster keyword that provides access to resources and config.
+        """
+        gpratk_settings = context.resources.dataset_settings.gridpathratoolkit
+        ds = context.resources.datastore
+
+        parts = {
+            "daily_weather": {
+                "func": _extract_csv,
+                "condition": gpratk_settings.daily_weather,
+            },
+            "aggregated_extended_solar_capacity": {
+                "func": _extract_capacity_factor,
+                "condition": "solar" in gpratk_settings.technology_types
+                and "extended" in gpratk_settings.processing_levels,
+            },
+            "aggregated_extended_wind_capacity": {
+                "func": _extract_capacity_factor,
+                "condition": "wind" in gpratk_settings.technology_types
+                and "extended" in gpratk_settings.processing_levels,
+            },
+            "solar_capacity_aggregations": {
+                "func": _extract_csv,
+                "condition": "solar" in gpratk_settings.technology_types
+                and (
+                    "extended" in gpratk_settings.processing_levels
+                    or "aggregated" in gpratk_settings.processing_levels
+                ),
+            },
+            "wind_capacity_aggregations": {
+                "func": _extract_csv,
+                "condition": "wind" in gpratk_settings.technology_types
+                and (
+                    "extended" in gpratk_settings.processing_levels
+                    or "aggregated" in gpratk_settings.processing_levels
+                ),
+            },
+        }
+
+        if part not in parts:
+            raise ValueError(f"Unable to process {part}!")
```

Comment:
This feels a bit messy to me, but it works, and the interface it provides via the settings yaml file is simple.

Reply:
hm i feel like this whole `parts` dictionary (maybe with the exception of the `func` should live in the settings classes. and then you can do what most of the setting classes do an verify the parts based on the working partitions.

Reply:
I'm not sure what you're thinking of here. What additional work are you saying should be done in the settings class? I'm not trying to verify that we've gotten valid settings here, I'm trying to control what processing happens based on the values found in the settings.

Reply:
- migrate logic about parts dict's `condition` into settings class. class will the compile the list of parts and verify that all of those parts are `working_partitions`.
- `gpratk_settings` in this function will only need to access `parts` (which is to say the parts you want to process not the working partitions)
- in `raw_gridpathratoolkit_assets` you'll still loop over the working_partitions, but in this factory, you will check in `gpratk_settings` if you are processing that part, if yes: process. if no, return empty df
- in this function, compile the list of parts to process via `_extract_csv` vs `_exctract_X`